main.cpp

```diff
@@ -0,0 +1,184 @@
+        : Nume(nume), Specie(specie), Rasa(rasa), CategorieDeProductie(categorieDeProductie), Greutate(greutate), Varsta(varsta) {}
+
+};
+
+
+int showMenu()
+{
+    int number;
+    cout << "Alegeti una dintre optiuni:" << endl;
+    cout << "\t 1) Adaugati animal" << endl;
+    cout << "\t 2) Afisati lista animalelor" << endl;
+    cout << "\t 3) Editati detaliile unui animal" << endl;
+    cout << "\t 4) Eliminati un animal" << endl;
+    cout << "\t 5) Iesiti din aplicatie" << endl;
+    cin >> number;
+    return number;
+}
+
+// Function to convert a string to lowercase
+string toLowerCase(const string& str)
+{
+    string lowerStr = str;
+    transform(lowerStr.begin(), lowerStr.end(), lowerStr.begin(), ::tolower);
+    return lowerStr;
+}
+
+
+// Helper function to find an animal by name
+//vector<Animal>::iterator findAnimalByName(vector<Animal>& animals, const string& name)
+//{
+//return find_if(animals.begin(), animals.end(), [&name](const Animal& a) { return a.Nume == name; });
+//}
+
+// Helper function to find an animal by name (case-insensitive)
+vector<Animal>::iterator findAnimalByName(vector<Animal>& animals, const string& name)
+{
+    string lowerName = toLowerCase(name);
+    return find_if(animals.begin(), animals.end(), [&lowerName](const Animal& a) { return toLowerCase(a.Nume) == lowerName; });
+}
+
+int main()
+{
+    vector<Animal> animals;
+    int choice;
+    Animal a;
+    string animalDeCautat;
+    string animalDeSters;
+    do
+    {
+        cout << "Ferma contine " << animals.size() << " animale." << endl;
+        choice = showMenu();
+        switch(choice)
+        {
+            case 1:
+            {
+                cout << "Ati ales sa adaugati un animal nou. Va rugam introduceti urmatoarele detalii: " << endl;
+                cout << "Nume: ";
+                cin >> a.Nume;
+                cout << "Specie: ";
+                cin >> a.Specie;
+                cout << "Rasa: ";
+                cin >> a.Rasa;
+                cout << "Categorie de productie: ";
+                cin >> a.CategorieDeProductie;
+                cout << "Greutate: ";
+                cin >> a.Greutate;
+                cout << "Varsta: ";
+                cin >> a.Varsta;
+
+                animals.push_back(a);
+            }
+            break;
+
+            case 2: // Afiseaza intreaga lista a animalelor
+                {
+                    if(animals.size() != 0)
+                    {
+                        for(int i = 0; i < animals.size(); i ++)
+                        cout << "Nume: " << animals[i].Nume << " " << "Specie: " << animals[i].Specie << " " << "Categorie de productie: " << animals[i].CategorieDeProductie
+                        << " " << "Greutate: " << animals[i].Greutate << " " << "Varsta: " << animals[i].Varsta << endl;
+                    }
+                    else
+                        cout << "Niciun animal intordus!" << endl;
+
+            break;
+                }
```

Comment:
aici ar fi mai elegant sa scoti toate astea intr-o functie separata si doar sa apelezi functia in switch